packages/server/api/src/app/flows/step-run/step-run-service.ts

```diff
@@ -50,14 +51,18 @@ export const stepRunService = {
     const endTime = performance.now();
     const duration = endTime - startTime;
 
-    sendWorkflowTestBlockEvent({
-      flowId: flowVersion.flowId,
-      success: result.success,
-      projectId,
-      duration,
-      userId,
-      step,
-    });
+    const trackEventsEnabled = await userService.getTrackEventsConfig(userId);
+    sendWorkflowTestBlockEvent(
```

Comment:
instead of passing trackEventsEnabled, I think we can just do an IF here and not call sendWorkflowTestBlockEvent